BREAKING CHANGE: support fastify v4

Allain55 opened the pull request:
This PR adds support for Fastify v4

Allain55 committed:
feat: support fastify v4

Add support for Fastify version 4.
BREAKING CHANGE: Fastify version 3 is no longer supported

dnlup commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@dnlup/fastify-traps",
-  "version": "1.2.0",
+  "version": "2.0.0",
```
There is no need to change this, it will be changed automatically by the release flow in the CI. I order to do so, there must be a 'BREAKING CHANGE: description' line in the commit body. A commit example:

```
feat: support fastify v4

Add support for Fastify version 4.
BREAKING CHANGE: Fastify version 3 is no longer supported
```

If you could amend the commit message like this it would be awesome 😉

Allain55 replied:
Hi @dnlup,
Thank you!
Indeed, Fastify v4 dropped the support of Node 12.
I've amended the commit, is there anything else I missed?

dnlup replied:
Thank you for changing the message. A couple of things are missing:

*  the `version` field in `package.json` should remain `1.2.0` (it is handled automatically by the release process)
* we can remove `12.x` from [this matrix in the tests](https://github.com/dnlup/fastify-traps/blob/574a980501d8b8b3a5a6f616fbeec04ff6332d64/.github/workflows/tests.yml#L19)

Allain55 replied:
Thank you!
Yeah, I didn't think it's necessary to revert the version in package.json. I pushed the requested changes

dnlup reviewed (changes requested):
Thank you @Allain55 and welcome! I left a few comments 😉. I have to see if Node version 12 is supported by fastify 4, it doesn't seem like it (tests are failing). In case it isn't we can remove it from the matrix I guess.

Allain55 committed:
feat: revert version to 1.2.0
feat: remove Node 12 from the testing matrix

BREAKING CHANGE: drop Node 12 support

dnlup reviewed (approved):
Thank you @Allain55

dnlup commented:
I'll try to release it tomorrow 😉
